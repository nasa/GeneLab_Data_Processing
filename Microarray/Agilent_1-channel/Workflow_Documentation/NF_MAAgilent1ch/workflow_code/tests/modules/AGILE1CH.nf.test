nextflow_process {

    name "Test Process AGILE1CH"
    script "modules/AGILE1CH.nf"
    process "AGILE1CH"
    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                limit_biomart_query = 6000
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("./bin/Agile1CMP.qmd")
                input[1] = file("../examples/runsheet/GLDS-367_microarray_v0_runsheet.csv")
                input[2] = file("https://figshare.com/ndownloader/files/36597120")
                input[3] = "Homo sapiens"
                input[4] = params.limit_biomart_query
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.de, process.out.versions).match()
        }
    }
}