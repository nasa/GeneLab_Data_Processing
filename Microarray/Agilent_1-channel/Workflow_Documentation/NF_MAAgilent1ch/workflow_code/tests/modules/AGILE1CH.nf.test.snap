{
    "Should run without failures": {
        "content": [
            [
                [
                    "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                    "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                    "differential_expression.csv:md5,f864a8c4ea88da4ee5af987f560a46bc",
                    "visualization_output_table.csv:md5,bcc4bf70f3c6bd93a58018c3dfb2719e",
                    "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                    "normalized_expression.csv:md5,1454f5a7e0df30d4aaf7fdc8d624bcb9",
                    "raw_intensities.csv:md5,3d01079abeeb126dff85e27c78f16760"
                ]
            ],
            [
                "versions.yml:md5,b1ef2efdecce6f0b2ce9340b2eb0b380"
            ]
        ],
        "timestamp": "2023-02-23T22:49:55+0000"
    }
}